docs/0.1/uploading_smart_contracts.md

```diff
@@ -0,0 +1,254 @@
+   contract will read from and write to).
+
+   The example smart contract has a manifest with the following contents:
+
+    ``` yaml
+    name: sawtooth_xo
+    version: '1.0'
+    inputs:
+      - '5b7349'
+    outputs:
+      - '5b7349'
+    ```
+
+1. Package the smart contract WASM file and the manifest in a smart contract
+   archive (a tar file with the `.scar` extension).
+
+    ``` console
+    $ tar -jcvf mycontract_0.1.0.scar mycontract.wasm manifest.yaml
+    ```
+
+    **IMPORTANT**: The name of the scar file must use the format
+    <code><i>name</i>_<i>version</i></code> (for example, `mycontract_0.1.0`).
+    This name corresponds to the contract argument used with the
+    `scabbard contract upload` command in a later step.
+
+### Connect to the scabbard-cli Container
+
+{:start="3"}
+
+3. Start a bash session in your node’s `scabbard-cli` Docker container (such as
+   `scabbard-cli-alpha`).  You will use this container to run the `scabbard`
+   commands to upload and configure the smart contract.
+
+   ```
+   $ docker exec -it scabbard-cli-alpha bash
+   root@scabbard-cli-alpha:/#
+   ```
+
+1. To simplify entering the `scabbard` commands in this procedure, set
+   `SERVICE_ID` to the fully qualified service ID for the scabbard service on
+   this circuit (such as `01234-ABCDE::gsAA`).
+
+   ```
+   root@scabbard-cli-alpha:/# export SERVICE_ID=01234-ABCDE::gsAA
+   ```
+
+### Download the Smart Contract Package
+
+{:start="5"}
+
+5. Download the scar file (such as `xo_0.4.2.scar`) to the `scabbard-cli`
+   container.
+
+   ```
+   root@scabbard-cli-alpha:/# curl -OLsS https://files.splinter.dev/scar/xo_0.4.2.scar
+   ```
+
+### Create a Contract Registry
+
+Each smart contract requires a contract registry that specifies the contract's
+name and owner. You must create a contract registry before you can upload the
+smart contract to the circuit.
+
+{:start="6"}
+
+6. Create a contract registry for the new smart contract.
+
+   This example creates a smart contract registry for `sawtooth_xo`, with the
+   Grid daemon as the owner (the `gridd` public key is used as the owner ID).
+   The Grid daemon's private key is used to sign this transaction.
+
+   ```
+   root@scabbard-cli-alpha:/# scabbard cr create sawtooth_xo \
+   --owners $(cat /root/.splinter/keys/gridd.pub) \
+   -k gridd \
+   -U 'http://splinterd-alpha:8085' \
+   --service-id $SERVICE_ID
+   ```
+
+### Upload the Smart Contract
+
+{:start="7"}
+
+7. Upload the smart contract to the Splinter circuit by specifying the smart
+   contract name and version (such as `xo:0.4.2`) and the path to the scar file.
+
+   ```
+   root@scabbard-cli-alpha:/# scabbard contract upload xo:0.4.2 \
+   --path . \
+   -k gridd \
+   -U 'http://splinterd-alpha:8085' \
+   --service-id $SERVICE_ID
+   ```
+
+   **IMPORTANT**: The smart contract name and version must match the scar file
+   name (with `:` instead of `_`). You can use wildcard characters in place
```

Comment:
"Wildcard characters" isn't super accurate, I'd say any valid semver version requirement (with the link for more details).